synapse/lib/lmdblayer.py

```diff
@@ -51,8 +53,15 @@ class LmdbLayer(s_layer.Layer):
         maxsize = self.conf.get('lmdb:maxsize')
         growsize = self.conf.get('lmdb:growsize')
 
+        # First check hive configuration.  If not set, use passed-in parameter (that defaults to False)
+        lockmemoryconf = self.conf.get('lmdb:lockmemory')
```

Comment:
if we went with the 'dedicated' conf idea from above, we could possibly simplify some of this logic since we wouldn't have a variable value coming in from arguments vs. configuration data.